### PR TITLE
Allow scheduled / delayed deletes

### PR DESCRIPTION
**Pull request by Saladoc**
This PR introduces a `Deletable` interface which takes over the existing deletion methods and introduces new ones for the scheduled deletion. Basically this is a do-over of the PR made by @cap5lut, incorporating the ideas from the discussion in #630.

I suggest either one of this and #630 to be merged and the other closed.

### CODE REVIEW

**Review by KILLEliteMaste — CHANGES_REQUESTED** [review]

**Review by Mysterypotatoguy — CHANGES_REQUESTED** [review]
Looks good from the API side, just a couple of entities missing for `Deletable`:
`Server`, `Role`, `MessageSet`(?)

n.b. `MessageSet` current implements as `deleteAll()` so unsure if this should be included

**Comment by Saladoc** [comment]
`MessageSet` should not be `Deleteable` as per the interface since it does not represent a singular deletable entity, but a collection of them. The other ones I will update.

**Review by KILLEliteMaste — CHANGES_REQUESTED** [review]

File: javacord-core/src/main/java/org/javacord/core/entity/server/ServerImpl.java [hunk 1]
```diff
@@ -1486,6 +1486,14 @@ public CompletableFuture<Void> delete() {
                 .execute(result -> null);
     }
 
+    @Override
+    public CompletableFuture<Void> delete(String reason) {
+        // Since the delete endpoint for servers does not support the audit log reason
+        // we silently discard it. Even if it was supported, it would be lost instantly
+        // as the server is deleted.
+        return delete();
+    }
```
[the remark this answers is not in the document]
Comment (Saladoc) [reply javacord-core/src/main/java/org/javacord/core/entity/server/ServerImpl.java]: Applied and tested those changes, discord does not complain about the header.

**Review by KILLEliteMaste — APPROVED** [review]
LGTM!

**Comment by Saladoc** [comment]
reopening after accidental branch deletion

**Review by Mysterypotatoguy — APPROVED** [review]

**Pull request merged**